Source/ShoutFactory.swift

```diff
@@ -0,0 +1,244 @@
+      $0.sizeToFit()
+    }
+
+    if imageView.image == nil { Dimensions.textOffset = 18 }
+
+    displayTimer.invalidate()
+    displayTimer = NSTimer.scheduledTimerWithTimeInterval(announcement.duration,
+      target: self, selector: "displayTimerDidFire", userInfo: nil, repeats: false)
+    setupFrames()
+  }
+
+  public func shout(to controller: UIViewController) {
+    guard let controller = controller.navigationController else { fatalError("The controller must contain a navigation bar") }
+
+    controller.view.addSubview(self)
+
+    frame = CGRect(x: 0, y: 0, width: UIScreen.mainScreen().bounds.width, height: 0)
+    backgroundView.frame = CGRect(x: 0, y: 0, width: Dimensions.width, height: 0)
+    UIView.animateWithDuration(0.35, animations: {
+      self.frame.size.height = Dimensions.height
+      self.backgroundView.frame.size.height = self.frame.height
+    })
+  }
+
+  // MARK: - Setup
+
+  public func setupFrames() {
+    backgroundView.frame = CGRect(x: 0, y: 0, width: Dimensions.width, height: Dimensions.height)
+    blurView.frame = backgroundView.bounds
+    gestureContainer.frame = CGRect(x: 0, y: Dimensions.height - 20, width: Dimensions.width, height: 20)
+    indicatorView.frame = CGRect(x: (Dimensions.width - Dimensions.indicatorWidth) / 2,
+      y: Dimensions.height - Dimensions.indicatorHeight - 5, width: Dimensions.indicatorWidth, height: Dimensions.indicatorHeight)
+    imageView.frame = CGRect(x: Dimensions.imageOffset, y: (Dimensions.height - Dimensions.imageSize) / 2 + 5,
+      width: Dimensions.imageSize, height: Dimensions.imageSize)
+    titleLabel.frame.origin = CGPoint(x: Dimensions.textOffset, y: imageView.frame.origin.y + 3)
+    subtitleLabel.frame.origin = CGPoint(x: Dimensions.textOffset, y: CGRectGetMaxY(titleLabel.frame) + 2.5)
+
+    if let text = subtitleLabel.text where text.isEmpty {
+      titleLabel.center.y = imageView.center.y - 2.5
+    }
+
+    [titleLabel, subtitleLabel].forEach {
+      $0.frame.size.width = Dimensions.width - Dimensions.imageSize - (Dimensions.imageOffset * 2) }
+  }
+
+  // MARK: - Actions
+
+  public func silent() {
+    UIView.animateWithDuration(0.35, animations: {
+      self.frame.size.height = 0
+      self.backgroundView.frame.size.height = self.frame.height
+      }, completion: { finished in
+        self.displayTimer.invalidate()
+        self.removeFromSuperview()
+    })
+  }
+
+  // MARK: - Timer methods
+
+  public func displayTimerDidFire() {
+    shouldSilent = true
+
+    guard !panGestureActive else { return }
+    silent()
+  }
+
+  // MARK: - Gesture methods
+
+  public func handleTapGestureRecognizer() {
+    guard let announcement = announcement else { return }
+    announcement.action?()
+    silent()
+  }
+
+  public func handlePanGestureRecognizer() {
+    let translation = panGestureRecognizer.translationInView(self)
+    var duration: NSTimeInterval = 0
+
+    if panGestureRecognizer.state == .Changed || panGestureRecognizer.state == .Began {
```

Comment:
Could you do `[UIGestureRecognizerState.Changed, UIGestureRecognizerState.Began].contains(panGestureRecognizer.state)` ?